src/objects.c

```diff
@@ -1555,53 +1555,263 @@ const char *p11prov_obj_get_ec_group_name(P11PROV_OBJ *obj)
     return (const char *)attr->pValue;
 }
 
+struct obj_export_ec_ctx {
+    BN_CTX *bnctx;
+    void *allocs[6];
```

Comment:
Not a huge fan of very opaque structures like this, do I understand correctly that this is done just to keep a reference to the allocated temporary values until the export is complete?

Why not simply go through params matching and freeing the buffer for specific keys so that this whole export_ctx becomes unnecessary ?

Reply:
Done as you suggested.